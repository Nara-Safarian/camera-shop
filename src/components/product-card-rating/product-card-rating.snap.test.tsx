import { render } from '@testing-library/react';
import ProductCardRating from './product-card-rating';

const REVIEWS = [{
  'id': '9309e98a-11e5-4b7c-a6d2-73b1d8e2f0bf',
  'userName': 'Дарья',
  'advantage': 'Цена соответствует качеству.',
  'disadvantage': 'Без объектива',
  'review': 'Это моя первая камера. Я в восторге, нареканий нет',
  'rating': 4,
  'createAt': '2023-03-28T11:52:11.598Z',
  'cameraId': 1
},
{
  'id': '6b4ce4f7-efd0-4798-985c-a357fab7ec79',
  'userName': 'Артём',
  'advantage': 'Недорогая, за такую цену отличный вариант.',
  'disadvantage': 'Трудно найти чехол. Заводские крайне дррогие',
  'review': 'Это моя первая камера. Я в восторге, нареканий нет',
  'rating': 2,
  'createAt': '2023-03-07T11:52:11.601Z',
  'cameraId': 1
}];

const RATING = 10;

test('ProductCardRating snapshot should be rendered correctly without reviews', () => {
  const {container} = render(
    <ProductCardRating reviewCount={RATING} />
  );
  expect(container).toMatchSnapshot();
});


test('ProductCardRating snapshot should be rendered correctly with reviews', () => {
  const {container} = render(
    <ProductCardRating reviewCount={RATING} reviews={REVIEWS} />
  );
  expect(container).toMatchSnapshot();
});
