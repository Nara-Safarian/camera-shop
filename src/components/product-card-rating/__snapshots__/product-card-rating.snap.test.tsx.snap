// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardRating snapshot should be rendered correctly with reviews 1`] = `
<div>
  <div
    class="rate product-card__rate"
  >
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-full-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-full-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-full-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <p
      class="visually-hidden"
    >
      Рейтинг: 
      3
    </p>
    <p
      class="rate__count"
    >
      <span
        class="visually-hidden"
      >
        Всего оценок:
      </span>
      10
    </p>
  </div>
</div>
`;

exports[`ProductCardRating snapshot should be rendered correctly without reviews 1`] = `
<div>
  <div
    class="rate product-card__rate"
  >
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <svg
      aria-hidden="true"
      height="16"
      width="17"
    >
      <use
        xlink:href="#icon-star"
      />
    </svg>
    <p
      class="visually-hidden"
    >
      Рейтинг: 
      0
    </p>
    <p
      class="rate__count"
    >
      <span
        class="visually-hidden"
      >
        Всего оценок:
      </span>
      10
    </p>
  </div>
</div>
`;
